packages/monorepo-builder/packages/init/src/Command/InitCommand.php

```diff
@@ -80,7 +83,32 @@ protected function execute(InputInterface $input, OutputInterface $output): int
      */
     private function getMonorepoBuilderVersion(): string
     {
-        $version = new Version(str_replace('x-dev', '0', $this->composer->getPackage()->getPrettyVersion()));
+        $version = null;
+
+        try {
+            $version = new Version(str_replace('x-dev', '0', $this->composer->getPackage()->getPrettyVersion()));
+        } catch (InvalidVersionException $e) {
+            // Version might not be explicitly set inside composer.json, looking for "vendor/composer/installed.json"
+            $installedJsonFilename = sprintf('%s/composer/installed.json', dirname(__DIR__, 6));
+
+            if (is_file($installedJsonFilename)) {
+                $installedJson = json_decode(NetteFileSystem::read($installedJsonFilename));
+
+                foreach ($installedJson as $installedPackage) {
+                    if ('symplify/monorepo-builder' === $installedPackage->name) {
+                        $version = new Version($installedPackage->version);
+
+                        break;
+                    }
+                }
+            }
+
+            if (null === $version) {
+                $this->symfonyStyle->warning('Could not update "symplify/monorepo-builder" version, please update it manually inside the root "composer.json" file');
+
+                return '%UPDATE_VERSION_HERE%';
+            }
+        }
```

Comment:
Such long code should be decoupled to own method, e.g. `getVersion()`

Reply:
Yes, you're right, I hope that my latest commit can solve this.